src/library/vm/vm_array.cpp

```diff
@@ -17,6 +18,13 @@ struct vm_array : public vm_external {
     virtual void dealloc() override { this->~vm_array(); get_vm_allocator().deallocate(sizeof(vm_array), this); }
     virtual vm_external * ts_clone(vm_clone_fn const &) override;
     virtual vm_external * clone(vm_clone_fn const &) override { lean_unreachable(); }
+    virtual unsigned int hash() override {
+        unsigned int h = 9435;
```

Comment:
I'm surprised there is no hashing functions for arrays or lists yet.  But we're missing many basic things.  Magic number seems randomly chosen, I'm happy!